worker/rhpv2.go

```diff
@@ -583,17 +690,17 @@ func (s *Session) Delete(ctx context.Context, sectorIndices []uint64, price type
 //
 // Note that it is typically not necessary to explicitly unlock a contract; the
 // host will do so automatically when the connection closes.
-func (s *Session) Unlock() (err error) {
+func (s *Session) Unlock(ctx context.Context) (err error) {
 	defer wrapErr(&err, "Unlock")
 	s.revision = rhpv2.ContractRevision{}
 	s.key = nil
-	return s.transport.WriteRequest(rhpv2.RPCUnlockID, nil)
+	return s.transport.writeRequest(ctx, rhpv2.RPCUnlockID, nil)
 }
 
 // Close gracefully terminates the session and closes the underlying connection.
-func (s *Session) Close() (err error) {
+func (s *Session) Close(ctx context.Context) (err error) {
```

Comment:
arg is unused

IMO `Transport.Close` should internally set a write deadline of like 1 second. We *want* to close gracefully (by sending `loopExit`), but if the write buffer is backed up, we really do not care about forcibly closing the connection.